ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/QueryFilterNodeTest.java

```diff
@@ -1676,4 +1688,277 @@ public void shouldThrowNotKeyColumnForBetween() {
     // Then:
     assertThat(e.getMessage(), containsString("A comparison must directly reference a key column"));
   }
+
+  @Test
+  public void shouldReturnNonKeyConstraintIntGreater() {
+    // Given:
+    final Expression expression = new ComparisonExpression(
+            Type.GREATER_THAN,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K")),
+            new IntegerLiteral(1)
+    );
+    QueryFilterNode filterNode = new QueryFilterNode(
+            NODE_ID,
+            source,
+            expression,
+            metaStore,
+            ksqlConfig,
+            false,
+            plannerOptions
+    );
+
+    // When:
+    final List<LookupConstraint> keys = filterNode.getLookupConstraints();
+
+    // Then:
+    assertThat(keys.size(), is(1));
+    assertThat(keys.get(0), instanceOf(NonKeyConstraint.class));
+  }
+
+  @Test
+  public void shouldReturnKeyConstraintInt() {
+    // Given:
+    when(plannerOptions.getTableScansEnabled()).thenReturn(true);
+    final Expression keyExp1 = new ComparisonExpression(
+            Type.GREATER_THAN,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K")),
+            new IntegerLiteral(1)
+    );
+    final Expression keyExp2 = new ComparisonExpression(
+            Type.EQUAL,
+            new UnqualifiedColumnReferenceExp(ColumnName.of("K")),
+            new IntegerLiteral(3)
+    );
+    final Expression expression = new LogicalBinaryExpression(
+            LogicalBinaryExpression.Type.AND,
+            keyExp1,
+            keyExp2
+    );
+    QueryFilterNode filterNode = new QueryFilterNode(
+            NODE_ID,
+            source,
+            expression,
+            metaStore,
+            ksqlConfig,
+            false,
+            plannerOptions);
+
+    // When:
+    final List<LookupConstraint> keys = filterNode.getLookupConstraints();
+
+    // Then:
+    assertThat(keys.size(), is(1));
+    assertThat(keys.get(0), instanceOf(KeyConstraint.class));
+    final KeyConstraint keyConstraint = (KeyConstraint) keys.get(0);
+    assertThat(keyConstraint.getKey(), is(GenericKey.genericKey(3)));
```

Comment:
Want to check the operator to show that you have the equals as well?  Finding a three probably indicates that the one you got, but good to be sure.

Reply:
Done